contracts/poap/src/contract.rs

```diff
@@ -0,0 +1,443 @@
+
+    // Create the cw721 message to send to mint the poap
+    let mint_msg = Cw721ExecuteMsg::Mint(MintMsg::<String> {
+        token_id: poap_id.to_string(),
+        owner: recipient_addr.to_string(),
+        token_uri: Some(event_info.base_poap_uri),
+        extension: event_info.event_uri,
+    });
+
+    let cw721_address = CW721_ADDRESS.load(deps.storage)?;
+    let wasm_execute_mint_msg = wasm_execute(cw721_address, &mint_msg, vec![])?;
+
+    // Update the next poap id state
+    let new_poap_id = poap_id + 1;
+    NEXT_POAP_ID.save(deps.storage, &new_poap_id)?;
+    // Save the new mint count for the sender's address
+    let new_recipient_mint_count = recipient_mint_count + 1;
+    MINTER_ADDRESS.save(
+        deps.storage,
+        recipient_addr.clone(),
+        &new_recipient_mint_count,
+    )?;
+
+    Ok(Response::new()
+        .add_attribute("action", action)
+        .add_attribute("sender", info.sender)
+        .add_attribute("recipient", recipient_addr.to_string())
+        .add_attribute("poap_id", poap_id.to_string())
+        .add_message(wasm_execute_mint_msg))
+}
+
+fn execute_update_event_info(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    start_time: Timestamp,
+    end_time: Timestamp,
+) -> Result<Response, ContractError> {
+    let mut event_info = EVENT_INFO.load(deps.storage)?;
+
+    // Check that is the event creator that is changing the event time frame
+    if event_info.creator != info.sender {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    // Check that the event is not ended
+    if event_info.is_ended(&env.block.time) {
+        return Err(ContractError::EventTerminated {
+            current_time: env.block.time,
+            end_time: event_info.end_time,
+        });
+    }
+
+    // Check that the event is not started
+    if event_info.is_started(&env.block.time) {
+        return Err(ContractError::EventStarted {
+            current_time: env.block.time,
+            start_time: event_info.start_time,
+        });
+    }
+
+    // Check that the start time is before the end time
+    if start_time.ge(&end_time) {
+        return Err(ContractError::StartTimeAfterEndTime {
+            start: start_time,
+            end: end_time,
+        });
+    }
+
+    // Check that the start time is not already passed
+    if env.block.time.ge(&start_time) {
+        return Err(ContractError::StartTimeBeforeCurrentTime {
+            current_time: env.block.time,
+            start_time,
+        });
+    }
+
+    // Check that the end time is not already passed
+    if env.block.time.ge(&end_time) {
+        return Err(ContractError::EndTimeBeforeCurrentTime {
+            current_time: env.block.time,
+            end_time,
+        });
+    }
+
+    // Update the event info
+    event_info.start_time = start_time;
+    event_info.end_time = end_time;
+    EVENT_INFO.save(deps.storage, &event_info)?;
```

Comment:
Can we use `EVENT_INFO.update` instead?

Reply:
Using update in this case is less efficent then just calling `save`. 
With update you are actualy calling another time `load`, then the lambda passed as argument that performs the modification and finaly `save`.  Since we have already loaded `EventInfo` we can just update the previosly loaded object and save it.